src/depth_packet_processor.cpp

```diff
@@ -98,4 +104,51 @@ void DepthPacketProcessor::setFrameListener(libfreenect2::FrameListener *listene
   listener_ = listener;
 }
 
+  DumpDepthPacketProcessor::DumpDepthPacketProcessor() : depth_frame(new Frame(512, 424, 4)), ir_frame(new Frame(512, 424,4)) {
+}
+
+DumpDepthPacketProcessor::~DumpDepthPacketProcessor(){
+  delete depth_frame;
+  delete ir_frame;
+}
+
+void DumpDepthPacketProcessor::process(const DepthPacket &packet) {
+    std::stringstream name;
+    name << packet.timestamp << "_" << packet.sequence << "_" << packet.buffer_length << ".depth";
+  
+    std::ofstream file(name.str().c_str());
+    file.write(reinterpret_cast<char *>(packet.buffer), packet.buffer_length);
+    file.close();
+
+    std::stringstream json;
+    json << "{" << std::endl;
+    json << "\"timestamp\": " << packet.timestamp << "," << std::endl;
+    json << "\"sequence\": " << packet.sequence << "," << std::endl;
+
+    timespec now;
+    clock_gettime(CLOCK_REALTIME, &now);
```

Comment:
This function is not portable. This is a pain point in the project so far. There is no portable timing function in C++98 unless you introduce a portability kludge.

We try to use glfwGetTime() or C++11 std::chrono depending on the build config, https://github.com/OpenKinect/libfreenect2/blob/master/src/logging.cpp#L182. Unfortunately this doesn't always work when the user disables OpenGL and does not enable C++11 (some Mac users, some Odroid users).

So two choices for replacing this:

1. Introduce a portable timing function
2. Depend on the build config to use one of glfwGetTime() or C++11 std::chrono, and deal with it when both are unavailable.